### ISSUE TITLE
Redirecting output of process to a pipe

### ISSUE THREAD

**Issue by azertyalex** [tsaarni/cpp-subprocess#7]
I create an ffmpeg process which needs to write to `/tmp/virtmic`, how can I redirect the output?

Bash equivalent:
```bash
ffmpeg -re -i test.webm -f s16le -ar 16000 -ac 1 - > /tmp/virtmic
```

**Comment by azertyalex** [comment]
I managed to do this with following code
```c++
void start() {
    virt_mic = new std::ofstream("/tmp/virtmic", std::ios::out | std::ios::binary);

    std::thread t([=, this]() {
      cmd = new subprocess::popen(
          "ffmpeg",
          {"-re", "-i", "-", "-f", "s16le", "-ar", "48000", "-ac", "1", "-"},
          *virt_mic);

      while (true) {
        if (!packets.empty()) {
          cmd->stdin().write(packets.front().data, packets.front().size);
          delete packets.front().data;
          packets.pop();
        }
      }
    });
    t.detach();
  };
`

**Comment by tsaarni** [comment]
Cool, great that you sorted it out 👍
